runtime/near-vm-runner/src/preload.rs

```diff
@@ -41,27 +54,76 @@ pub struct ContractCallPrepareResult {
 
 pub struct ContractCaller {
     pool: ThreadPool,
-    prepared: Vec<CallInner>,
+    vm_kind: VMKind,
+    vm_config: VMConfig,
+    vm_data_private: VMDataPrivate,
+    vm_data_shared: VMDataShared,
+    preloaded: Vec<CallInner>,
 }
 
 impl ContractCaller {
-    pub fn new(num_threads: usize) -> ContractCaller {
-        ContractCaller { pool: ThreadPool::new(num_threads), prepared: Vec::new() }
+    pub fn new(num_threads: usize, vm_kind: VMKind, vm_config: VMConfig) -> ContractCaller {
+        let (shared, private) = match vm_kind {
+            VMKind::Wasmer0 => (
+                VMDataShared::Wasmer0,
+                VMDataPrivate::Wasmer0(if SHARE_MEMORY_INSTANCE {
+                    Some(
+                        WasmerMemory::new(
+                            vm_config.limit_config.initial_memory_pages,
+                            vm_config.limit_config.max_memory_pages,
+                        )
+                        .unwrap(),
+                    )
+                } else {
+                    None
+                }),
+            ),
+            VMKind::Wasmer1 => {
+                let store = default_wasmer1_store();
+                let store_clone = store.clone();
+                (
+                    VMDataShared::Wasmer1(store),
+                    VMDataPrivate::Wasmer1(if SHARE_MEMORY_INSTANCE {
+                        Some(
+                            Wasmer1Memory::new(
+                                &store_clone,
+                                vm_config.limit_config.initial_memory_pages,
+                                vm_config.limit_config.max_memory_pages,
+                            )
+                            .unwrap(),
+                        )
+                    } else {
+                        None
+                    }),
+                )
+            }
+            _ => panic!("Not currently supported"),
```

Comment:
```suggestion
            VMKind::Wasmer => panic!("Not currently supported"),
```

This way, if we add a new `VMKind` like `Wasmer2`, we'll get a compile time error, which would give us a chance to consider if  we should panic or if we should just implement this. 

In general, `_` top-level patters should be avoided.

Reply:
Ack.